### ISSUE TITLE
run-make with target wasm32-unknown-unknown has hard-to-diagnose errors

### ISSUE THREAD

**Issue by pnkfelix** [rust-lang/rust#109132]
I tried this code:

```toml
# config.toml

[build]
target = ["x86_64-unknown-linux-gnu","wasm32-unknown-unknown"]
```

Bootstrap invocation:
```shell
% python x.py build    --stage 1   library/std  && python x.py test    --stage 1    tests/run-make
```

I expected to see this happen: the test suite runs and gives me successes or failures relevant to my Rust project source, or gives me diagnostics that tell me about unsatisified dependencies.

Instead, this happened: Mysterious run-make test failures like "rust-lld not found" or "foo.js not found"

----

Each of the failures are easy to work around.

E.g. rust-lld can be satisfied by adding:

```
[rust]
lld = true
```

to the `config.toml`.

And I *think* (but have not yet confirmed) that `foo.js not found` is addressed by setting:

```toml
[build]
nodejs = "node"
```

(and ensuring that node is indeed installed).

But the experience is pretty bad in terms of running the `run-make` tests. It might be worth considering adding a prerequisite to the relevant wasm run-make tests that does a trivial dummy run of the basics, and if it fails, spits out an error telling the developer what changes they should make if they want to do `run-make` tests with this target enabled.

**Comment by jyn514** [comment]
i agree for the specific case with node here we should double check it in sanity.rs, but in general the makefiles can have arbitrary dependencies and there's not a whole ton we can do (imagine a test that assume it has gnu parallel and doesn't do error checking).

cc https://github.com/rust-lang/rust/issues/40713
